app/uk/gov/hmrc/apprenticeshiplevy/connectors/DesConnector.scala

```diff
@@ -58,38 +63,72 @@ trait EmployerDetailsEndpoint extends Timer with Logging {
-      audit(new ALAEvent("readEmprefDetails", empref)) {
-        val headers = createDesHeaders
-        des.httpClient.GET[HodDesignatoryDetailsLinks](url, Seq(), headers).flatMap { response =>
-          val details = DesignatoryDetails(Some(empref))
-          response.links.map { links =>
-            logger.debug((links.employer ++ links.communication).mkString(" "))
-            val getEmployer = links.employer.map(getDetails(_)).getOrElse(Future.successful(None))
-            val getComms = links.communication.map(getDetails(_)).getOrElse(Future.successful(None))
-            for {
-              e <- getEmployer
-              c <- getComms
-            } yield (details.copy(employer=e, communication=c))
-          }.getOrElse(Future.successful(details))
+      audit(ALAEvent("readEmprefDetails", empref)) {
+        des.httpClient.GET[Either[UpstreamErrorResponse, HodDesignatoryDetailsLinks]](
+          url         = url,
+          queryParams = Seq(),
+          headers     = createDesHeaders
+        ) flatMap {
+          case Right(response) =>
+            response.links.map {
+              links =>
+                val employer: Future[Option[DesignatoryDetailsData]] =
+                  links
+                    .employer
+                    .map(getDetails(_))
+                    .getOrElse(Future.successful(None))
+                val communication: Future[Option[DesignatoryDetailsData]] =
+                  links
+                    .communication
+                    .map(getDetails(_))
+                    .getOrElse(Future.successful(None))
+
+                employer.flatMap {
+                  emp =>
+                    communication.flatMap {
+                      com =>
+                        Future.successful(details.copy(employer = emp, communication = com))
+                    }
+                }.recoverWith {
+                  case e: Throwable =>
+                    logger.error(s"getDetails call failed: ${e.getMessage}")
+                    Future.successful(details)
+                }
+            }.getOrElse(Future.successful(details))
+          case Left(e: UpstreamErrorResponse) =>
+            if (e.statusCode == BAD_REQUEST)
+              throw new BadRequestException(e.message)
+            if (e.statusCode == NOT_FOUND)
+              throw new NotFoundException(e.message)
+            else
+              throw UpstreamErrorResponse(e.message, e.statusCode, e.reportAs)
         }
       }
     }
   }
 
-  protected def getDetails(path: String)(implicit hc: HeaderCarrier): Future[Option[DesignatoryDetailsData]] = {
-    val headers = createDesHeaders
-    des.httpClient.GET[DesignatoryDetailsData](s"${des.baseUrl}${path}", Seq(), headers).map { data => Some(data) }.recover(errorHandler)
-  }
-
-  protected val errorHandler: PartialFunction[Throwable, Option[DesignatoryDetailsData]] = {
-        case e => {
-          // $COVERAGE-OFF$
```

Comment:
will this cause beef with there being no matching tag for the on below? otherwise maybe delete the other tag

Reply:
Good spot, have fixed it